zha/application/platforms/light/__init__.py

```diff
@@ -787,20 +803,32 @@ def __init__(
                 effect_list.append(EFFECT_COLORLOOP)
                 if self._color_cluster_handler.color_loop_active == 1:
                     self._effect = EFFECT_COLORLOOP
-        self._external_supported_color_modes = supported_color_modes = (
-            filter_supported_color_modes(self._supported_color_modes)
-        )
-        if len(supported_color_modes) == 1:
-            self._color_mode = next(iter(supported_color_modes))
-        else:  # Light supports color_temp + hs, determine which mode the light is in
-            assert self._color_cluster_handler
+
             if (
-                self._color_cluster_handler.color_mode
-                == Color.ColorMode.Color_temperature
+                self._color_cluster_handler.color_mode is not None
+                and self._color_cluster_handler.color_mode in ZCL_TO_ENTITY_COLOR_MODE
             ):
-                self._color_mode = ColorMode.COLOR_TEMP
+                self._color_mode = ZCL_TO_ENTITY_COLOR_MODE[
+                    self._color_cluster_handler.color_mode
+                ]
+
+        self._external_supported_color_modes = filter_supported_color_modes(
+            self._supported_color_modes
+        )
+
+        if self._color_mode == ColorMode.UNKNOWN:
+            if len(self._external_supported_color_modes) == 1:
+                self._color_mode = next(iter(self._external_supported_color_modes))
             else:
-                self._color_mode = ColorMode.XY
+                # Light supports color_temp + hs, determine which mode the light is in
```

Comment:
The comment seems to be outdated (before and now). This is a leftover from when HA only provided HS and we converted it to XY manually.

```suggestion
                # Light supports color_temp + xy, determine which mode the light is in
```


(Also, it seems like we completely ignore the light possibly being in HS color mode here. But seeing as it's an opt-in option, I'd not bother with it for now.
I'd just get rid of HS support entirely sooner or later, since all colors are "broken" there and Z2M also doesn't offer it for HA: just converts HS to XY to send to light, which we also do by default.)